packages/gitbook/src/lib/links.ts

```diff
@@ -82,7 +82,9 @@ export function baseUrl(): string {
  */
 export function absoluteHref(href: string, withHost: boolean = false): string {
     const base = withHost ? baseUrl() : basePath();
-    return `${base}${href.startsWith('/') ? href.slice(1) : href}`;
+    const absoluteHref = `${base}${href.startsWith('/') ? href.slice(1) : href}`;
+    // Trim trailing slashes
+    return absoluteHref.replace(/\/$/, '');
```

Comment:
<img width="732" alt="Screenshot 2024-11-26 at 17 19 26" src="https://github.com/user-attachments/assets/13648b38-2ca1-4771-a4a3-7b2623af9e5b">

We pass hrefs to `absoluteHref` (meaning path + query params); but this logic is checking the whole thing as a path.

It's not very clean and error prone. We should improve this code to handle the fact that it's not just a path

Reply:
Sorry I made a safer approach for that, only doing this in canonical.